pkg/controller/defaultdomaincert/default_domain_cert_controller_test.go

```diff
@@ -0,0 +1,875 @@
+	secret, err := reconciler.getSecret(ddc)
+
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "failed to get key from store")
+	assert.Nil(t, secret)
+}
+
+func TestGetSecret_CertificateContentIsNil(t *testing.T) {
+	mockStore := newMockStore()
+	// Set files to exist but with nil content
+	mockStore.files[testCertPath] = nil
+	mockStore.shouldExist[testCertPath] = true
+	mockStore.setFileContent(testKeyPath, []byte(testKeyContent))
+
+	reconciler := createTestReconciler(nil, mockStore)
+
+	ddc := createTestDefaultDomainCertificate("test-ddc", testNamespace, testSecretName)
+
+	secret, err := reconciler.getSecret(ddc)
+
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "failed to get certificate from store")
+	assert.Nil(t, secret)
+}
+
+func TestGetSecret_KeyContentIsNil(t *testing.T) {
+	mockStore := newMockStore()
+	mockStore.setFileContent(testCertPath, []byte(testCertContent))
+	// Set key file to exist but with nil content
+	mockStore.files[testKeyPath] = nil
+	mockStore.shouldExist[testKeyPath] = true
+
+	reconciler := createTestReconciler(nil, mockStore)
+
+	ddc := createTestDefaultDomainCertificate("test-ddc", testNamespace, testSecretName)
+
+	secret, err := reconciler.getSecret(ddc)
+
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "failed to get key from store")
+	assert.Nil(t, secret)
+}
+
+func TestGetSecret_EmptyNamespace(t *testing.T) {
+	mockStore := newMockStore()
+	mockStore.setFileContent(testCertPath, []byte(testCertContent))
+	mockStore.setFileContent(testKeyPath, []byte(testKeyContent))
+
+	reconciler := createTestReconciler(nil, mockStore)
+
+	ddc := createTestDefaultDomainCertificate("test-ddc", "", testSecretName)
+
+	secret, err := reconciler.getSecret(ddc)
+
+	require.NoError(t, err)
+	require.NotNil(t, secret)
+
+	assert.Equal(t, testSecretName, secret.Name)
+	assert.Equal(t, "", secret.Namespace)
+}
+
+func TestNewReconciler_AddCertFileError(t *testing.T) {
+	mockStore := newMockStore()
+	mockStore.addFileErr = errors.New("failed to add cert file")
+
+	conf := &config.Config{
+		DefaultDomainCertPath: testCertPath,
+		DefaultDomainKeyPath:  testKeyPath,
+	}
+
+	err := NewReconciler(conf, &testutils.FakeManager{}, mockStore)
+
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "adding default domain cert")
+	assert.Contains(t, err.Error(), "failed to add cert file")
+}
+
+func TestNewReconciler_AddKeyFileError(t *testing.T) {
+	mockStore := newMockStore()
+	// Set specific error for the key file (second AddFile call)
+	mockStore.addFileKeyErr = errors.New("failed to add key file")
+
+	conf := &config.Config{
+		DefaultDomainCertPath: testCertPath,
+		DefaultDomainKeyPath:  testKeyPath,
+	}
+
+	err := NewReconciler(conf, &testutils.FakeManager{}, mockStore)
+
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "adding default domain key")
+	assert.Contains(t, err.Error(), "failed to add key file")
+}
+
+func TestReconcile_SecretAlreadyExists_UpdatesExistingSecret(t *testing.T) {
```

Comment:
can you colocate test reconcile tests/tests for each function

Reply:
done